plugins/utils/todo-comments.nix

```diff
@@ -1,290 +1,429 @@
-      extraPackages = [ cfg.ripgrepPackage ];
-      extraConfigLua = ''
-        require("todo-comments").setup${helpers.toLuaObject setupOptions}
-      '';
+  settingsExample = {
+    highlight = {
+      pattern = [
+        ".*<(KEYWORDS)\s*:"
+        ".*<(KEYWORDS)\s*"
+      ];
+    };
+  };
 
-      keymaps = flatten (
-        mapAttrsToList (
+  extraOptions = {
+    keymaps =
+      let
+        mkKeymapOption =
           optionName: funcName:
-          let
-            keymap = cfg.keymaps.${optionName};
-
-            cwd = optionalString (keymap.cwd != null) " cwd=${keymap.cwd}";
-            keywords = optionalString (keymap.keywords != null) " keywords=${keymap.keywords}";
-          in
-          optional (keymap != null) {
-            mode = "n";
-            inherit (keymap) key options;
-            action = ":${funcName}${cwd}${keywords}<CR>";
-          }
-        ) commands
-      );
-
-      # Automatically enable plugins if keymaps have been set
-      plugins = mkMerge [
-        (mkIf (cfg.keymaps.todoTrouble != null) { trouble.enable = true; })
-        (mkIf (cfg.keymaps.todoTelescope != null) { telescope.enable = true; })
-      ];
+          mkCompositeOption "Keymap settings for the `:${funcName}` function." {
+            key = mkOption {
+              type = types.str;
+              default = null;
+              description = "Key for the `${funcName}` function.";
+            };
+
+            cwd = mkOption {
+              type = types.nullOr types.str;
+              description = "Specify the directory to search for comments";
+              default = null;
+              example = "~/projects/foobar";
+            };
+
+            keywords = mkOption {
+              type = types.nullOr types.str;
+              description = ''
+                Comma separated list of keywords to filter results by.
+                Keywords are case-sensitive.
+              '';
+              default = null;
+              example = "TODO,FIX";
+            };
+
+            options = keymaps.mapConfigOptions;
+          };
+      in
+      mapAttrs mkKeymapOption keymapsActions;
+
+    ripgrepPackage = mkPackageOption {
+      name = "ripgrep";
+      default = pkgs.ripgrep;
     };
+  };
+
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion =
+          cfg.keymaps.todoTelescope != null && cfg.keymaps.todoTelescope.key != null
+          -> config.plugins.telescope.enable;
+        message = ''
+          Nixvim: You have enabled the `telescope` integration with git-worktree.
+          However, you have not enabled the `telescope` plugin itself (`plugins.telescope.enable = true`).
+        '';
+      }
+      {
+        assertion =
+          cfg.keymaps.todoTrouble != null && cfg.keymaps.todoTrouble.key != null
+          -> config.plugins.trouble.enable;
+        message = ''
+          Nixvim: You have enabled the `trouble` integration with todo-comments.
```

Comment:
```suggestion
          Nixvim(plugins.todo-comments): You have enabled todo-comment's `trouble` integration.
```

Reply:
IDK if the `(plugins.todo-comments)` is really needed since it's obvious from the assertion text, be we _usually_ include something like it in our warnings/assertions?